File: Python/Task12/n18495.py
# https://inf-ege.sdamgia.ru/problem?id=18495

s = '1'*40 + '2'*40
while '111' in s:
	s = s.replace('111', '2', 1)
	s = s.replace('222', '1', 1)
print(s)

File: C++/Task12/n18495/n18495.cpp
// https://inf-ege.sdamgia.ru/problem?id=18495

#include <iostream>

using namespace std;

int main()
{
	string s = string(40, '1') + string(40, '2');
	while (s.find("111") != string::npos)
	{
		size_t pos = s.find("111");
		s.replace(pos, 3, "2");
		pos = s.find("222");
		s.replace(pos, 3, "1");
	}
	std::cout << s << std::endl;
	return 0;
}